explainers/hidden-content-explainer.md

```diff
@@ -200,24 +205,87 @@ representing the active match is collapsed or it has a style which makes it
 invisible, the scroll will be canceled. Otherwise, the target match will be
 scrolled into view.
 
-### Alternatives Considered
+## Accessibility
+
+`content-visibility: hidden-matchable` subtrees will not be included in the
+accessibility tree because they are not visible on the screen. This is the same
+as `content-visibility: hidden` and offscreen/unrendered `content-visibility:
+auto` subtrees.
+
+The content inside `content-visibility: hidden-matchable` subtrees will still
+remain accessible to assisstive technology users because there should always be
+a way to reveal the content besides using find-in-page, such as a button to
+expand the content like a `<details>` element.
+
+## Scroll timing and element fragment issue
+
+With this feature, find-in-page and ScrollToTextFragment will have an
+asynchronous flow for firing the beforematch event and scrolling when the match
+is in a `content-visibility: hidden-matchable` subtree:
+1. A match within a `content-visibility: hidden-matchable` subtree has been
+   found
+2. Wait for the next requestAnimationFrame
+3. Fire the beforematch event on the `hidden-matchable` ancestor
+4. Wait for the next requestAnimationFrame
+5. Scroll to the target match
+
+This behavior was implemented in response to early feedback about pages where
+the beforematch event handler queued an asynchronous task to remove the
+`hidden-matchable` property, in which case the requestAnimationFrame gap between
+firing the beforematch event and scrolling was needed in order to correctly
+scroll to the revealed content.
+
+Supporting beforematch for element fragments is problematic because scrolling
+for element fragments is always done synchronously. To be more specific,
+consider the following code:
+```html
+<div style="height:100px"></div>
+<div id=foo></div>
+<script>
+window.scrollY; // returns 0
+window.location.hash = '#foo';
+window.scrollY; // returns 100
+</script>
+```
+Since changes to the element fragment trigger synchronous scrolling, which we
+can't change without a breaking the current behavior, we will have to fire the
+beforematch event and then scroll synchronously, instead of the
+rAF->beforematch->rAF->scroll behavior we have for ScrollToTextFragment and
+find-in-page. This means that pages which need a full rAF in between the
+beforematch event and the scroll in order to reveal the content to scroll to may
+not be able to reveal the target content in time for the scroll.
+
+A possible mitigation for this would be to check to see if the target element is
```

Comment:
I think this is an unnecessary complication. If script is changing window.location.hash then it can also reveal content without the need for the beforematch handler (although we'll still run one after the scroll)

Reply:
I agree - I just think it would be better if we are stuck in the case where we post a task to fire beforematch for all cases. I'll remove it since we are already talking about doing it sync